Penumbra support v2

hdevalence opened the pull request:
Replaces #1124 but from the Penumbra fork to make collaboration easier

Commit:
wip: penumbra relayer provider

remove copied-over cosmos provider tests, rename processor cosmos -> penumbra

rename ccp -> pcp

reformat into new relayer dir structure

update penumbra types to point to buf.build

building again

agouin committed:
Penumbra buf go gen

agouin committed:
Use go prefix override

jtieri committed:
fix penumbra

jtieri committed:
fix: implement MsgSubmitMisbehaviour

jtieri committed:
fix: remove unnecessary proto file + fix msg type cast

jtieri committed:
chore: add removal of penumbra client protos in protocgen

hdevalence committed:
working penumbra relayer functionality

wip: unbase64?

wip: multiple messages per penumbra tx

wip: stub SendMessages impl

wip: attempt to split out common method

wip: use random anchor and work around path renaming

wip: improve logging

wip: changes during pairing

https://www.youtube.com/watch?v=RYonSOkZ5ZE

clean up logs

skip height bug workaround

drop debug panic

avahowell committed:
update penumbra chain processor connection and channel message processing

conorsch commented:
Using the latest code in this branch, along with the configs in https://github.com/penumbra-zone/penumbra/tree/fa2f750f4ec049bc0d7195dc368fc7490f5c347e/deployments/relayer , we're able to build a working path:

```
❯ rly paths list
 0: penumbra_path        -> chns(✔) clnts(✔) conn(✔) (penumbra-testnet-pasiphae<>penumbra-testnet-pasiphae-5d99ea66)
 ```

jtieri commented on relayer/client.go:
```diff
@@ -510,6 +510,7 @@ func findMatchingClient(ctx context.Context, src, dst *Chain, newClientState ibc
 // parseClientIDFromEvents parses events emitted from a MsgCreateClient and returns the
 // client identifier.
 func parseClientIDFromEvents(events []provider.RelayerEvent) (string, error) {
+	fmt.Printf("client.go events: %+v\n", events)
```
looks like an artifact from debugging, this could be removed

jtieri commented on relayer/channel.go:
```diff
@@ -74,7 +74,7 @@ func (c *Chain) CreateOpenChannels(
 		).
 		WithPathProcessors(pp).
-		WithInitialBlockHistory(0).
+		WithInitialBlockHistory(100).
```
is there a specific reason for this change?

jtieri commented on relayer/events.go:
```diff
@@ -0,0 +1,56 @@
+// ParseConnectionIDFromEvents parses events emitted from a MsgConnectionOpenInit or
+// MsgConnectionOpenTry and returns the connection identifier.
+func ParseConnectionIDFromEvents(events []provider.RelayerEvent) (string, error) {
+	fmt.Printf("events.go events: %+v\n", events)
```
another print statement that could go

jtieri commented on relayer/chains/penumbra/penumbra_chain_processor.go:
```diff
@@ -0,0 +1,552 @@
+	"github.com/cosmos/relayer/v2/relayer/processor"
+	"github.com/cosmos/relayer/v2/relayer/provider"
+
+	//	jsonrpcclient "github.com/cometbft/cometbft/rpc/jsonrpc/client"
```
commented import statement could be removed

jtieri commented on relayer/chains/penumbra/penumbra_chain_processor.go:
```diff
@@ -0,0 +1,552 @@
+}
+
+/*
+func (pcp *PenumbraChainProcessor) queryCycle(ctx context.Context, persistence *queryCyclePersistence) error {
```
if this is no longer being used as a reference we could pull this out

jtieri commented on relayer/chains/penumbra/tx.go:
```diff
@@ -0,0 +1,2289 @@
+func parseEventsFromABCIResponse(resp abci.ResponseDeliverTx) []provider.RelayerEvent {
+	var events []provider.RelayerEvent
+
+	fmt.Printf("tx.go events: %+v\n", resp.Events)
```
another print statement that could be removed

jtieri commented on relayer/chains/penumbra/tx.go:
```diff
@@ -0,0 +1,2289 @@
+			Attributes: attributes,
+		})
+	}
+	fmt.Printf("parsed events: %+v\n", events)
```
here as well

jtieri commented on relayer/chains/penumbra/tx.go:
```diff
@@ -0,0 +1,2289 @@
+	if err != nil {
+		return nil, false, err
+	}
+	cc.log.Info("waiting for penumbra tx to commit", zap.String("syncRes", fmt.Sprintf("%+v", syncRes)))
```
can we capitalize the first word in this log statement? for reference we do have a style guide for logging which can be found [here](https://github.com/cosmos/relayer/blob/main/docs/logging.md)

jtieri commented on relayer/chains/penumbra/tx.go:
```diff
@@ -0,0 +1,2289 @@
+		if err != nil {
+			return err
+		}
+		cc.log.Info("got penumbra tx result", zap.String("res", fmt.Sprintf("%+v", res)))
```
See above comment

jtieri commented on relayer/chains/penumbra/tx.go:
```diff
@@ -0,0 +1,2289 @@
+var defaultUpgradePath = []string{"upgrade", "upgradedIBCState"}
+
+/*
+fn apphash_spec() -> ics23::ProofSpec {
```
this looks like it was being used to port some Rust code but if it's no longer needed it could be removed

jtieri reviewed (commented):
great work penumbra team 🔥 🔥 
just a handful of nits but otherwise this lgtm

conorsch committed:
cleanup logging statements for review

Responding to review comments, honoring the style guide for logging, and
removing some unused reference code that was commented out while
debugging.

conorsch commented:
Thanks for pointing me to the style guide, @jtieri. I've updated all the places you called out, and touched up a few more logging statements to be compliant. After making these changes, I rebuilt and reran the path build script, to ensure I didn't inadventerntly break something. Indeed, we're still good:

```
❯ rly paths list
 0: penumbra_path        -> chns(✔) clnts(✔) conn(✔) (penumbra-testnet-pasiphae<>penumbra-testnet-pasiphae-7549032f)
 ```

jtieri reviewed (approved):
🚢 🚢

jtieri committed:
Merge branch 'main' into penumbra-v2
